sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java

```diff
@@ -185,60 +162,48 @@ private static Object deserializeBody(String value, Type resultType, Type wireTy
     }
 
     /**
-     * Given:
-     * (1). the {@code java.lang.reflect.Type} (resultType) of java proxy method return value
-     * (2). and {@link ReturnValueWireType} annotation value indicating 'entity type' (wireType)
-     * of same REST API's wire response body
+     * Given: (1). the {@code java.lang.reflect.Type} (resultType) of java proxy method return value (2). and {@link
+     * ReturnValueWireType} annotation value indicating 'entity type' (wireType) of same REST API's wire response body
      * this method construct 'response body Type'.
      *
-     * Note: When {@link ReturnValueWireType} annotation is applied to a proxy method, then the raw
-     * HTTP response content will need to parsed using the derived 'response body Type' then converted
-     * to actual {@code returnType}.
+     * Note: When {@link ReturnValueWireType} annotation is applied to a proxy method, then the raw HTTP response
+     * content will need to parsed using the derived 'response body Type' then converted to actual {@code returnType}.
      *
      * @param resultType the {@code java.lang.reflect.Type} of java proxy method return value
      * @param wireType the {@code java.lang.reflect.Type} of entity in REST API response body
      * @return the {@code java.lang.reflect.Type} of REST API response body
      */
     private static Type constructWireResponseType(Type resultType, Type wireType) {
-        Objects.requireNonNull(resultType);
         Objects.requireNonNull(wireType);
-        //
-        Type wireResponseType = resultType;
 
-        if (resultType == byte[].class) {
-            if (wireType == Base64Url.class) {
-                wireResponseType = Base64Url.class;
-            }
-        } else if (resultType == OffsetDateTime.class) {
+        Type wireResponseType = resultType;
+        if (resultType == OffsetDateTime.class) {
             if (wireType == DateTimeRfc1123.class) {
                 wireResponseType = DateTimeRfc1123.class;
             } else if (wireType == UnixTime.class) {
                 wireResponseType = UnixTime.class;
             }
-        } else {
-            if (TypeUtil.isTypeOrSubTypeOf(resultType, List.class)) {
-                final Type resultElementType = TypeUtil.getTypeArgument(resultType);
-                final Type wireResponseElementType = constructWireResponseType(resultElementType, wireType);
-
-                wireResponseType = TypeUtil.createParameterizedType(
-                    (Class<?>) ((ParameterizedType) resultType).getRawType(), wireResponseElementType);
-            } else if (TypeUtil.isTypeOrSubTypeOf(resultType, Map.class) || TypeUtil.isTypeOrSubTypeOf(resultType,
-                Response.class)) {
```

Comment:
Removed `Response.class` for a few reasons:

- I don't think `Response` would be a sensible `resultType` to get here since we unbox `Response` to its parameterized type in an earlier check, so this would need something like `Response<Response<>>`.
- The `typeArguments` retrieval would fail most of the time here since `Response<T>` only has a single type argument in the interface class, only `ResponseBase<H, T>` or a custom type could have enough type arguments to work here.